### PR TITLE
Feature: adding refresh token function to oauth2 service

### PR DESCRIPTION
**Pull request by eapg**
* Adding refresh token function to oauth2 service
* Adding app_access_token_fixture
* Adding app_refresh_token_fixture
* Adding test to refresh expired access token

### CODE REVIEW

File: lib/services/oauth2_service.ex [hunk 1]
```diff
@@ -52,6 +52,49 @@ defmodule EaRestaurantDataLoader.Oauth2Service do
...
+    validate_client_credentials(client, client_secret, Status.active())
+
+    case Oauth2Util.validate_token(access_token, secret_key) do
+      {:ok, _} ->
+        %{
+          :client_name => client.client_name,
+          :access_token => access_token,
+          :refresh_token => refresh_token,
+          :scopes => client_scope.scope,
+          :expire_in => client.access_token_expiration_time
```
Comment (reynoldmorel) [inline lib/services/oauth2_service.ex]: I think `expire_in` should be `expires_in` and also I think it should be the current date - expiration date of the token in seconds, not `client.access_token_expiration_time`

File: lib/services/oauth2_service.ex [hunk 2]
```diff
@@ -52,6 +52,49 @@ defmodule EaRestaurantDataLoader.Oauth2Service do
...
+        }
+
+      {:error, _} ->
+        Oauth2Util.validate_token(refresh_token, secret_key)
```
Comment (reynoldmorel) [inline lib/services/oauth2_service.ex]: I think you should expect `{:ok,_}` from here in order to validate the token properly?

File: test/lib/service/oauth2_service_test.exs [hunk 3]
```diff
@@ -33,5 +36,59 @@ defmodule EaRestaurantDataLoader.Oauth2ServiceTest do
       assert login_response.expires_in == app_client.access_token_expiration_time
       assert login_response.scopes == scopes.scope
     end
+
+    test " refresh token when expired access token" do
```
Comment (reynoldmorel) [inline test/lib/service/oauth2_service_test.exs:40]: * Add a test when a new access token is not build.
* Add a test where the refresh token is not valid anymore

**Review by reynoldmorel — CHANGES_REQUESTED** [review]

File: lib/services/oauth2_service.ex (hunk 2 above)
Comment (eapg) [reply lib/services/oauth2_service.ex]: Done.

File: lib/services/oauth2_service.ex (hunk 1 above)
Comment (eapg) [reply lib/services/oauth2_service.ex]: Done.

File: test/lib/service/oauth2_service_test.exs (hunk 3 above)
Comment (eapg) [reply test/lib/service/oauth2_service_test.exs:40]: Done.

**Review by reynoldmorel — APPROVED** [review]

**Pull request merged**